src/FileInput/SelectedFile.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import propTypes from 'prop-types'
+
+import { colors, spacers } from '../theme.js'
+import { AttachFile } from '../icons/AttachFile.js'
+import { Loading } from '../icons/Status.js'
+
+const SelectedFile = ({
+    label,
+    onRemove,
+    removeText,
+    loading,
+    onCancel,
+    cancelText,
+}) => (
+    <p>
+        <span className="icon">{loading ? <Loading /> : <AttachFile />}</span>
+
+        <span className="text">
+            <span className="label">{label}</span>
+
+            {loading && onCancel && cancelText && (
+                <span className="action" onClick={onCancel}>
+                    {cancelText}
+                </span>
+            )}
+
+            {!loading && (
+                <span className="action" onClick={onRemove}>
+                    {removeText}
+                </span>
+            )}
+        </span>
+
+        <style jsx>{`
+            p {
+                display: flex;
+                margin: 0;
+                padding-top: ${spacers.dp8};
+                line-height: 14px;
```

Comment:
Should be `line-height: 18px`.

I can see that @cooper-joe has not included line-heights in his design specs (please do that @cooper-joe).
But I can see a difference between the story and the design (the stories also seem quite dense in this regard)

Reply:
Thanks for pointing that out to me. You comment has reminded me of what I agreed with Joe: some of the spacing in the specs was not really clear to me, so I would ask him to review once I was done...
@cooper-joe : could you have a look [here](https://deploy-preview-384--dhis2-ui-core.netlify.com/?path=/story/fileinput--design-system-stacking-order) to see what I need to change about the spacing

Reply:
Two changes:
- Padding under button increase +4px
- Padding between help/error and the file list +4px (not an increase in line height in the actual file list)

The line heights are correct, it's just those paddings between 'sections' of the component. I know what you mean with the specs looking a little different/less dense than in browser, I think this has a lot to do with browser/OS font smoothing. I've just measured pixel for pixel and these were the only changes haha.

Reply:
@cooper-joe could you also add line-height values to the specs? This will def. help in the future to use the correct values.
Btw, it's generally a bad idea to use
(in-depth info: https://iamvdo.me/en/blog/css-font-metrics-line-height-and-vertical-align)

* font-size to determine an elements height
font-size does not determine the height of an element, the font itself does (if there's no explicit line-height value).

* use the same value for both font-size and line-height
When using the same line-height as font-size, you might end up with touching characters.
We can solve this with padding of course, but as long as we have control over the font, we can also do this with line-height (which we should use anyways) without having to use padding